CMakeLists.txt

```
@@ -132,22 +108,78 @@ celeritas_setup_option(CELERITAS_CORE_GEO ORANGE)
 celeritas_setup_option(CELERITAS_CORE_GEO Geant4 _allow_g4)
 celeritas_define_options(CELERITAS_CORE_GEO "Celeritas runtime geometry")
 
-if(NOT CELERITAS_DEBUG OR CELERITAS_USE_VecGeom)
-  set(_default_lock ON)
-else()
-  set(_default_lock OFF)
+#----------------------------------------------------------------------------#
+# CMAKE VERSION CHECKS
+#----------------------------------------------------------------------------#
+
+if(CMAKE_VERSION VERSION_LESS 3.13 AND CELERITAS_USE_CUDA AND CELERITAS_USE_MPI)
+  message(FATAL_ERROR "Celeritas requires CMake 3.13 or higher "
+    "when building with CUDA + MPI.")
+endif()
+if(CMAKE_VERSION VERSION_LESS 3.18 AND CELERITAS_USE_CUDA
+    AND CELERITAS_USE_VecGeom)
+  message(FATAL_ERROR "VecGeom+CUDA requires CMake 3.18 or higher to support "
+    "\"Separable compilation\".")
+endif()
+if(CMAKE_VERSION VERSION_LESS 3.18 AND CMAKE_CUDA_ARCHITECTURES)
+  message(FATAL_ERROR "The CMAKE_CUDA_ARCHITECTURES flag is not compatible "
+    "with this version of CMake. Set CMAKE_CUDA_FLAGS.")
+endif()
+if(CMAKE_VERSION VERSION_LESS 3.22 AND CELERITAS_USE_HIP)
+  message(WARNING "HIP support is immature; CMake 3.22+ is recommended.")
 endif()
-cmake_dependent_option(CELERITAS_TEST_RESOURCE_LOCK
-  "Only run one GPU-enabled test at a time" "${_default_lock}"
-  "CELERITAS_BUILD_TESTS" OFF
-)
 
-if(NOT DEFINED CMAKE_CXX_STANDARD)
-  # Default CMake standard to C++17 for this directory and underneath
-  set(CMAKE_CXX_STANDARD 17)
-  set(CMAKE_CXX_EXTENSIONS OFF)
+#----------------------------------------------------------------------------#
+# CMAKE INTRINSIC OPTIONS
+#
+# These are generally used to initialize properties on targets, and it's
+# possible Celeritas is being built inside another project. Instead of saving
+# these as cache variables (which change the defaults project-wide, including
+# changing behavior of other code that was loaded before Celeritas) set them as
+# local variables to be inherited underneath Celeritas.
+#----------------------------------------------------------------------------#
+
+### Configuration ###
+celeritas_set_default(CMAKE_EXPORT_NO_PACKAGE_REGISTRY ON)
+celeritas_set_default(CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY ON)
+celeritas_set_default(CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY ON)
```

According to https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY.html#variable:CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY this is deprecated since cmake 3.16
(same for `CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY`
```suggestion
celeritas_set_default(CMAKE_FIND_USE_PACKAGE_REGISTRY FALSE)
celeritas_set_default(CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY FALSE)
```